lib/improver/percentile.py

```diff
@@ -111,39 +113,17 @@ def process(self, cube):
         """
         # Store data type and enforce the same type on return.
         data_type = cube.dtype
+        # Test that collapse coords are present in cube before proceding.
+        if sum([test_coord in coord.name() for coord in cube.coords()
```

Comment:
This if test is a bit long, which makes it a bit difficult to read. Could you split this up a little?